src/headerssync.cpp

```diff
@@ -102,11 +102,13 @@ HeadersSyncState::ProcessingResult HeadersSyncState::ProcessPresync(const
         return ret;
     }
 
-    if (full_headers_message || m_state == State::REDOWNLOAD) {
-        // A full headers message means the peer may have more to give us;
-        // also if we just switched to REDOWNLOAD then we need to re-request
+    if (m_state == State::REDOWNLOAD) {
+        // If we just switched to REDOWNLOAD then we need to re-request
```

Comment:
6d9c992cb92d5ce24e49fe13167f211b43e95ef9

Not sure I follow, the previous commit was already huge and hard to follow, why was this one broken out and not included there? Or that one broken to smaller changes?

"Break apart logic" - commit message is still drafty, probably on purpose, right? Seems like a "split" instead of a "break", but maybe the diff is only in my head :)